src/main/groovy/lang/ObjectRange.java

```diff
@@ -306,14 +303,13 @@ public int size() {
                 }
             } else {
                 // let's brute-force calculate the size by iterating start to end
-                final Iterator iter = new StepIterator(this, 1);
```

Comment:
We're not in a critical section here, and uncontended synchronization is pretty cheap anyway.